src/Features/Core/EditAndContinue/IEditAndContinueWorkspaceService.cs

```diff
@@ -12,6 +12,9 @@ internal interface IEditAndContinueWorkspaceService : IWorkspaceService
         EditSession EditSession { get; }
         DebuggingSession DebuggingSession { get; }
 
+        event EventHandler<DebuggingStateChangedEventArgs> BeforeDebuggingStateChanged;
+        void OnBeforeDebuggingStateChanged(DebuggingState before, DebuggingState after);
```

Comment:
I know you use it for tests, but I don't love having this method be in the interface.  It seems like it should be in the implementation, and just the event should be in the interface.